### ISSUE TITLE
[Windows] R2R Interop_COM._NETClients_Primitives Failure

### ISSUE THREAD

**Issue by echesakovMSFT** [dotnet/coreclr#20986]
For example, here
https://ci.dot.net/job/dotnet_coreclr/job/master/job/x86_release_windows_nt_r2r/2395/
https://ci.dot.net/job/dotnet_coreclr/job/master/job/x64_checked_windows_nt_r2r/2678/

```
Return code:      1
  Raw output file:      D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\Reports\Interop.COM\NETClients\Primitives\NETClientPrimitives\NETClientPrimitives.output.txt
  Raw output:
  BEGIN EXECUTION
                1 file(s) copied.
        " D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\Tests\Core_Root\crossgen.exe" /Platform_Assemblies_Paths D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\Tests\Core_Root;D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\Interop\COM\NETClients\Primitives\NETClientPrimitives\IL;D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\Interop\COM\NETClients\Primitives\NETClientPrimitives /in NETClientPrimitives.org /out NETClientPrimitives.exe
        Microsoft (R) CoreCLR Native Image Generator - Version 4.5.30319.0
        Copyright (c) Microsoft Corporation.  All rights reserved.
        
        Native image NETClientPrimitives.exe generated successfully.
         "D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\Tests\Core_Root\corerun.exe" NETClientPrimitives.exe 
        Numeric RNG seed: 37
        Test Failure: System.Runtime.InteropServices.COMException (0x80040154): Retrieving the COM class factory for component with CLSID {53169A33-E85D-4E3C-B668-24E438D0929B} failed due to the following error: 80040154 Class not registered (Exception from HRESULT: 0x80040154 (REGDB_E_CLASSNOTREG)).
           at NetClient.NumericTests..ctor(Int32 seed) in D:\j\workspace\x64_release_w---f2b03790\tests\src\Interop\COM\NETClients\Primitives\NumericTests.cs:line 22
           at NetClient.Program.Main(String[] doNotUse) in D:\j\workspace\x64_release_w---f2b03790\tests\src\Interop\COM\NETClients\Primitives\Program.cs:line 21
        Expected: 100
        Actual: 101
        END EXECUTION - FAILED
        FAILED
        Test Harness Exitcode is : 1
        
  To run the test:
  > set CORE_ROOT=D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\Tests\Core_Root
  > D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\Interop\COM\NETClients\Primitives\NETClientPrimitives\NETClientPrimitives.cmd
  
        Expected: True
        Actual:   False
        Stack Trace:
          D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\TestWrappers\Interop.COM\Interop.COM.XUnitWrapper.cs(391,0): at Interop_COM._NETClients_Primitives_NETClientPrimitives_NETClientPrimitives_._NETClients_Primitives_NETClientPrimitives_NETClientPrimitives_cmd()
        
  
  Return code:      1
  Raw output file:      D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\Reports\Interop.COM\NETClients\IDispatch\NETClientIDispatch\NETClientIDispatch.output.txt
  Raw output:
  BEGIN EXECUTION
                1 file(s) copied.
        " D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\Tests\Core_Root\crossgen.exe" /Platform_Assemblies_Paths D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\Tests\Core_Root;D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\Interop\COM\NETClients\IDispatch\NETClientIDispatch\IL;D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\Interop\COM\NETClients\IDispatch\NETClientIDispatch /in NETClientIDispatch.org /out NETClientIDispatch.exe
        Microsoft (R) CoreCLR Native Image Generator - Version 4.5.30319.0
        Copyright (c) Microsoft Corporation.  All rights reserved.
        
        Native image NETClientIDispatch.exe generated successfully.
         "D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\Tests\Core_Root\corerun.exe" NETClientIDispatch.exe 
        Test Failure: System.Runtime.InteropServices.COMException (0x80040154): Retrieving the COM class factory for component with CLSID {0F8ACD0C-ECE0-4F2A-BD1B-6BFCA93A0726} failed due to the following error: 80040154 Class not registered (Exception from HRESULT: 0x80040154 (REGDB_E_CLASSNOTREG)).
           at NetClient.Program.Validate_Numeric_In_ReturnByRef() in D:\j\workspace\x64_release_w---f2b03790\tests\src\Interop\COM\NETClients\IDispatch\Program.cs:line 19
           at NetClient.Program.Main(String[] doNotUse) in D:\j\workspace\x64_release_w---f2b03790\tests\src\Interop\COM\NETClients\IDispatch\Program.cs:line 165
        Expected: 100
        Actual: 101
        END EXECUTION - FAILED
        FAILED
        Test Harness Exitcode is : 1
        
  To run the test:
  > set CORE_ROOT=D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\Tests\Core_Root
  > D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\Interop\COM\NETClients\IDispatch\NETClientIDispatch\NETClientIDispatch.cmd
  
        Expected: True
        Actual:   False
        Stack Trace:
          D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\TestWrappers\Interop.COM\Interop.COM.XUnitWrapper.cs(321,0): at Interop_COM._NETClients_IDispatch_NETClientIDispatch_NETClientIDispatch_._NETClients_IDispatch_NETClientIDispatch_NETClientIDispatch_cmd()
      Interop_COM._NETClients_Primitives_NETClientPrimitives_NETClientPrimitives_._NETClients_Primitives_NETClientPrimitives_NETClientPrimitives_cmd [FAIL]
      Interop_COM._NETClients_IDispatch_NETClientIDispatch_NETClientIDispatch_._NETClients_IDispatch_NETClientIDispatch_NETClientIDispatch_cmd [FAIL]
      Interop_COM._NETClients_Aggregation_NETClientAggregation_NETClientAggregation_._NETClients_Aggregation_NETClientAggregation_NETClientAggregation_cmd [FAIL]
        
  
  Return code:      1
  Raw output file:      D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\Reports\Interop.COM\NETClients\Aggregation\NETClientAggregation\NETClientAggregation.output.txt
  Raw output:
  BEGIN EXECUTION
                1 file(s) copied.
        " D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\Tests\Core_Root\crossgen.exe" /Platform_Assemblies_Paths D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\Tests\Core_Root;D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\Interop\COM\NETClients\Aggregation\NETClientAggregation\IL;D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\Interop\COM\NETClients\Aggregation\NETClientAggregation /in NETClientAggregation.org /out NETClientAggregation.exe
        Microsoft (R) CoreCLR Native Image Generator - Version 4.5.30319.0
        Copyright (c) Microsoft Corporation.  All rights reserved.
        
        Native image NETClientAggregation.exe generated successfully.
         "D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\Tests\Core_Root\corerun.exe" NETClientAggregation.exe 
        Test Failure: System.Runtime.InteropServices.COMException (0x80040154): Retrieving the COM class factory for component with CLSID {4CEFE36D-F377-4B6E-8C34-819A8BB9CB04} failed due to the following error: 80040154 Class not registered (Exception from HRESULT: 0x80040154 (REGDB_E_CLASSNOTREG)).
           at NetClient.Program.ValidateNativeOuter() in D:\j\workspace\x64_release_w---f2b03790\tests\src\Interop\COM\NETClients\Aggregation\Program.cs:line 22
           at NetClient.Program.Main(String[] doNotUse) in D:\j\workspace\x64_release_w---f2b03790\tests\src\Interop\COM\NETClients\Aggregation\Program.cs:line 40
        Expected: 100
        Actual: 101
        END EXECUTION - FAILED
        FAILED
        Test Harness Exitcode is : 1
        
  To run the test:
  > set CORE_ROOT=D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\Tests\Core_Root
  > D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\Interop\COM\NETClients\Aggregation\NETClientAggregation\NETClientAggregation.cmd
  
        Expected: True
        Actual:   False
        Stack Trace:
          D:\j\workspace\x64_release_w---f2b03790\bin\tests\Windows_NT.x64.Release\TestWrappers\Interop.COM\Interop.COM.XUnitWrapper.cs(251,0): at Interop_COM._NETClients_Aggregation_NETClientAggregation_NETClientAggregation_._NETClients_Aggregation_NETClientAggregation_NETClientAggregation_cmd()
```

**Comment by echesakovMSFT** [comment]
@AaronRobinsonMSFT

**Comment by echesakovMSFT** [comment]
https://ci.dot.net/job/dotnet_coreclr/job/master/job/arm64_cross_checked_windows_nt_r2r_flow

**Comment by AaronRobinsonMSFT** [comment]
Addressed with #21010
